src/main/java/ch/njol/skript/expressions/ExprItemAmount.java

```diff
@@ -27,73 +27,75 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
 import ch.njol.skript.util.slot.Slot;
-import ch.njol.util.coll.CollectionUtils;
 import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
 import org.eclipse.jdt.annotation.Nullable;
 
 @Name("Item Amount")
 @Description("The amount of an <a href='classes.html#itemstack'>item stack</a>.")
 @Examples("send \"You have got %item amount of player's tool% %player's tool% in your hand!\" to player")
-@Since("2.2-dev24")
+@Since("2.2-dev24, 2.7 (itemstacks)")
 public class ExprItemAmount extends SimplePropertyExpression<Object, Long> {
-	
-    static {
-        register(ExprItemAmount.class, Long.class, "item[[ ]stack] (amount|size|number)", "slots/itemtypes");
-    }
 
+	static {
+		register(ExprItemAmount.class, Long.class, "item[[ ]stack] (amount|size|number)", "slots/itemtypes/itemstacks");
+	}
 	
 	@Override
 	public Long convert(final Object item) {
-    	return (long) (item instanceof ItemType ? ((ItemType) item).getAmount() : ((Slot) item).getAmount());
+		if (item instanceof ItemType) {
+			return (long) ((ItemType) item).getAmount();
+		} else if (item instanceof Slot) {
+			return (long) ((Slot) item).getAmount();
+		} else {
+			return (long) ((ItemStack) item).getAmount();
+		}
 	}
-	
+
 	@Override
-    public @Nullable Class<?>[] acceptChange(Changer.ChangeMode mode) {
-        return (mode != ChangeMode.REMOVE_ALL) ? CollectionUtils.array(Number.class) : null;
-    }
+	public @Nullable Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (mode != ChangeMode.REMOVE_ALL)
+			return new Class[]{Number.class};
+		return new Class[0];
+	}
```

Comment:
I remember getting asked this in a previous pr of mine, not 100% sure if just going with `mode != ChangeMode.REMOVAL_ALL` is a good choice, since if we add another ChangeMode if ever it'd be accepted by default without any proper support

Reply:
hmm this is how it was previously, your reasoning is sensible but idk tbh

Reply:
Agree, this would accept it by default which isn't a good choice therefore explicitly specifying the change modes is safer